engine/engine_impl.go

```diff
@@ -205,51 +205,64 @@ func (k *Kubernetes) Run(ctx context.Context, specv runtime.Spec, stepv runtime.
 	return k.waitForTerminated(ctx, spec, step)
 }
 
-func (k *Kubernetes) waitFor(ctx context.Context, spec *Spec, conditionFunc func(e watch.Event) (bool, error)) error {
+func (k *Kubernetes) waitFor(ctx context.Context, spec *Spec, conditionFunc func(pod *v1.Pod) (bool, error)) error {
 	label := fmt.Sprintf("io.drone.name=%s", spec.PodSpec.Name)
 	lw := &cache.ListWatch{
 		ListFunc: func(options metav1.ListOptions) (k8sruntime.Object, error) {
```

Comment:
I'm not too familiar with the underlying code, but just wanted to double-check and make sure it is thread safe to modify the `metav1.ListOptions`?

Reply:
I picked it up from the code in `kubectl`, I believe `Watch` process might set up some options we want to preserve when setting out label selector.